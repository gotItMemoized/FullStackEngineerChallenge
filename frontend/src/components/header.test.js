import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Header from './Header';
import 'jest-dom/extend-expect';

afterEach(cleanup);

it('renders without crashing', () => {
  const { container } = render(
    <MemoryRouter>
      <Header />
    </MemoryRouter>,
  );
  expect(container).toMatchSnapshot();
});

it('should not nav buttons if not logged in', () => {
  const { container } = render(
    <MemoryRouter>
      <Header />
    </MemoryRouter>,
  );
  expect(container).toMatchSnapshot();
  const innerDiv = document.getElementById('logout-btn');
  expect(innerDiv).toEqual(null);
  const innerDiv2 = document.getElementById('performance-menu-btn');
  expect(innerDiv2).toEqual(null);
});

it('should have some buttons if logged in as regular user', () => {
  const { container } = render(
    <MemoryRouter>
      <Header currentUser={{ loggedIn: true }} />
    </MemoryRouter>,
  );
  expect(container).toMatchSnapshot();
  const innerDiv = document.getElementById('logout-btn');
  expect(innerDiv).toHaveTextContent('Log out');
  const innerDiv2 = document.getElementById('performance-menu-btn');
  expect(innerDiv2).toHaveTextContent('Performance Reviews');
  const innerDiv3 = document.getElementById('user-menu-btn');
  expect(innerDiv3).toEqual(null);
});

it('should have more buttons if logged in as admin user', () => {
  const { container } = render(
    <MemoryRouter>
      <Header currentUser={{ loggedIn: true, isAdmin: true }} />
    </MemoryRouter>,
  );
  expect(container).toMatchSnapshot();
  const innerDiv = document.getElementById('logout-btn');
  expect(innerDiv).toHaveTextContent('Log out');
  const innerDiv2 = document.getElementById('performance-menu-btn');
  expect(innerDiv2).toHaveTextContent('Performance Reviews');
  const innerDiv3 = document.getElementById('users-menu-btn');
  expect(innerDiv3).toHaveTextContent('Users');
});
