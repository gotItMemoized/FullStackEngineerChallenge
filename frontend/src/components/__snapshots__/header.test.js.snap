// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <nav
    aria-label="main navigation"
    class="navbar"
    role="navigation"
  >
    <div
      class="navbar-brand"
    >
      <a
        aria-current="page"
        class="navbar-item active"
        href="/"
      >
        <strong>
          Coworker Yelp
        </strong>
      </a>
      <button
        aria-expanded="false"
        aria-label="menu"
        class="navbar-burger burger"
        data-target="navbarBasicExample"
        type="button"
      >
        <span
          aria-hidden="true"
        />
        <span
          aria-hidden="true"
        />
        <span
          aria-hidden="true"
        />
      </button>
    </div>
    <div
      class="navbar-menu undefined"
      id="navbarBasic"
    />
  </nav>
</div>
`;

exports[`should have more buttons if logged in as admin user 1`] = `
<div>
  <nav
    aria-label="main navigation"
    class="navbar"
    role="navigation"
  >
    <div
      class="navbar-brand"
    >
      <a
        aria-current="page"
        class="navbar-item active"
        href="/"
      >
        <strong>
          Coworker Yelp
        </strong>
      </a>
      <button
        aria-expanded="false"
        aria-label="menu"
        class="navbar-burger burger"
        data-target="navbarBasicExample"
        type="button"
      >
        <span
          aria-hidden="true"
        />
        <span
          aria-hidden="true"
        />
        <span
          aria-hidden="true"
        />
      </button>
    </div>
    <div
      class="navbar-menu undefined"
      id="navbarBasic"
    >
      <div
        class="navbar-start"
      >
        <a
          class="navbar-item"
          href="/users"
          id="users-menu-btn"
        >
          Users
        </a>
        <a
          class="navbar-item"
          href="/performance"
          id="performance-menu-btn"
        >
          Performance Reviews
        </a>
      </div>
      <div
        class="navbar-end"
      >
        <div
          class="navbar-item"
        >
          <div
            class="buttons"
          >
            <button
              class="button is-light"
              id="logout-btn"
              type="button"
            >
              Log out
            </button>
          </div>
        </div>
      </div>
    </div>
  </nav>
</div>
`;

exports[`should have some buttons if logged in as regular user 1`] = `
<div>
  <nav
    aria-label="main navigation"
    class="navbar"
    role="navigation"
  >
    <div
      class="navbar-brand"
    >
      <a
        aria-current="page"
        class="navbar-item active"
        href="/"
      >
        <strong>
          Coworker Yelp
        </strong>
      </a>
      <button
        aria-expanded="false"
        aria-label="menu"
        class="navbar-burger burger"
        data-target="navbarBasicExample"
        type="button"
      >
        <span
          aria-hidden="true"
        />
        <span
          aria-hidden="true"
        />
        <span
          aria-hidden="true"
        />
      </button>
    </div>
    <div
      class="navbar-menu undefined"
      id="navbarBasic"
    >
      <div
        class="navbar-start"
      >
        <a
          class="navbar-item"
          href="/performance"
          id="performance-menu-btn"
        >
          Performance Reviews
        </a>
      </div>
      <div
        class="navbar-end"
      >
        <div
          class="navbar-item"
        >
          <div
            class="buttons"
          >
            <button
              class="button is-light"
              id="logout-btn"
              type="button"
            >
              Log out
            </button>
          </div>
        </div>
      </div>
    </div>
  </nav>
</div>
`;

exports[`should not nav buttons if not logged in 1`] = `
<div>
  <nav
    aria-label="main navigation"
    class="navbar"
    role="navigation"
  >
    <div
      class="navbar-brand"
    >
      <a
        aria-current="page"
        class="navbar-item active"
        href="/"
      >
        <strong>
          Coworker Yelp
        </strong>
      </a>
      <button
        aria-expanded="false"
        aria-label="menu"
        class="navbar-burger burger"
        data-target="navbarBasicExample"
        type="button"
      >
        <span
          aria-hidden="true"
        />
        <span
          aria-hidden="true"
        />
        <span
          aria-hidden="true"
        />
      </button>
    </div>
    <div
      class="navbar-menu undefined"
      id="navbarBasic"
    />
  </nav>
</div>
`;
